addons/advanced_fatigue/functions/fnc_calculateBodyTemperature.sqf

```diff
@@ -0,0 +1,56 @@
+/*
+* Author: Pasrules
+* Calculates body temperature in degC, baseLine = 37
+*
+* Arguments:
+* 0: bodyTemperature <NUMBER>
+*
+* Return Value:
+* bodyTemperature <NUMBER>
+*
+*
+* Example
+* [37] call ace_advanced_fatigue_fnc_calculateBodyTemperature
+*/
+
+#include "script_component.hpp"
+params ["_bodyTemp"];
+
+if (["ACE_Weather"] call EFUNC(common,isModLoaded)) then {
+
+    //Weather
+    private _playerAltitude = (getPosASL ACE_player) select 2;
+    private _temperature = _playerAltitude call EFUNC(weather,calculateTemperatureAtHeight);
+    private _barometricPressure = _playerAltitude call EFUNC(weather,calculateBarometricPressure);
+    private _airDensity = [_temperature, _barometricPressure, humidity] call EFUNC(weather,calculateAirDensity);
+    private _windSpeed = [eyePos ACE_player, false, false, false] call EFUNC(weather,calculateWindSpeed);
+    private _windFactor = linearConversion [0, 20, _windSpeed, 0, 1, true];
+
+    //Body
+    private _factors = [_bodyTemp] call FUNC(calculateBodyFactors);
+
+    //Environment
+    private _sun = 120 * (1 - overcast);
+    private _air = (_temperature - MALE_TEMPERATURE) * E ^ _windFactor * _airDensity;
+    private _environment = (_sun + _air) * BODY_SURFACE_AREA;
+
+    //Heating, muscles produce 80% waste heat
+    private _speed = vectorMagnitude velocity player min 10;
```

Comment:
I think Pabo was saying that if you're falling or in a vehicle, your speed would technically be zero, since you're not exerting any effort to fall/sit.